tests/contrib/cherrypy/test_middleware.py

```diff
@@ -0,0 +1,429 @@
+        # ensure CherryPy uses the last set configuration to be sure
+        # there are no breaking changes for who uses `ddtrace-run`
+        # with the `TraceMiddleware`
+        assert cherrypy.tools.tracer._service == "test.cherrypy.service"
+        TraceMiddleware(
+            cherrypy,
+            self.tracer,
+            service="new-intake",
+        )
+        self.getPage("/")
+        time.sleep(0.01)  # Without this here, span may not be ready for inspection, and timings can be incorrect.
+
+        self.assertHeader("Content-Type", "text/html;charset=utf-8")
+        self.assertStatus("200 OK")
+
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 1
+
+        assert cherrypy.tools.tracer._service == "new-intake"
+
+    def test_child(self):
+        start = time.time()
+        self.getPage("/child")
+        time.sleep(0.01)  # Without this here, span may not be ready for inspection, and timings can be incorrect.
+        end = time.time()
+
+        # ensure request worked
+        self.assertStatus("200 OK")
+        self.assertHeader("Content-Type", "text/html;charset=utf-8")
+        self.assertBody("child")
+
+        # ensure trace worked
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 2
+
+        spans_by_name = {s.name: s for s in spans}
+
+        s = spans_by_name["cherrypy.request"]
+        assert s.span_id
+        assert s.trace_id
+        assert not s.parent_id
+        assert s.service == "test.cherrypy.service"
+        assert s.resource == "GET /child"
+        assert s.start >= start
+        assert s.duration <= end - start
+        assert s.error == 0
+
+        c = spans_by_name["child"]
+        assert c.span_id
+        assert c.trace_id == s.trace_id
+        assert c.parent_id == s.span_id
+        assert c.service == "test.cherrypy.service"
+        assert c.resource == "child"
+        assert c.start >= start
+        assert c.duration <= end - start
+        assert c.error == 0
+
+    def test_success(self):
+        start = time.time()
+        self.getPage("/")
+        time.sleep(0.01)  # Without this here, span may not be ready for inspection, and timings can be incorrect.
+        end = time.time()
+
+        # ensure request worked
+        self.assertStatus("200 OK")
+        self.assertHeader("Content-Type", "text/html;charset=utf-8")
+        self.assertBody("Hello world!")
+
+        # ensure trace worked
+        assert not self.tracer.current_span(), self.tracer.current_span().pprint()
+        spans = self.tracer.writer.pop()
+        assert len(spans) == 1
+        s = spans[0]
+        assert s.service == "test.cherrypy.service"
+        assert s.resource == "GET /"
+        assert s.start >= start
+        assert s.duration <= end - start
+        assert s.error == 0
+        assert_span_http_status_code(s, 200)
+        assert s.meta.get(http.METHOD) == "GET"
+
+        services = self.tracer.writer.pop_services()
```

Comment:
```suggestion
```
This was probably copy-paste from an older integration, we (thankfully) don't need to do this check anymore. (Same with the other tests that check it)

Reply:
Good to know! I've removed.